src/core/thread/link_quality.cpp

```diff
@@ -90,116 +72,136 @@ void LinkQualityInfo::AddRss(int8_t aNoiseFloor, int8_t aRss)
     // Multiply the the RSS value by a precision multiple (currently -8).
 
     newValue = static_cast<uint16_t>(-aRss);
-    newValue <<= kRssAveragePrecisionMultipleBitShift;
+    newValue <<= kPrecisionBitShift;
 
-    oldAverage = mRssAverage;
+    oldAverage = mAverage;
 
-    if (mCount >= kRssCountForWeightCoefficientOneEighth)
-    {
-        // New average = old average * 7/8 + new value * 1/8
-        mRssAverage = static_cast<uint16_t>(((oldAverage << 3) - oldAverage + newValue) >> 3);
-    }
-    else if (mCount >= kRssCountForWeightCoefficientOneFourth)
-    {
-        // New average = old average * 3/4 + new value * 1/4
-        mRssAverage = static_cast<uint16_t>(((oldAverage << 2) - oldAverage + newValue) >> 2);
-    }
-    else if (mCount >= kRssCountForWeightCoefficientOneHalf)
+    switch (mCount)
     {
+    case 0:
+        mAverage = newValue;
+        break;
+
+    case 1:
         // New average = old average * 1/2 + new value * 1/2
-        mRssAverage = (oldAverage + newValue) >> 1;
-    }
-    else
-    {
-        mRssAverage = newValue;
+        mAverage = (oldAverage + newValue) >> 1;
+        break;
+
+    case 2:
+        // New average = old average * 2/3 + new value * 1/3
+        mAverage = static_cast<uint16_t>(((oldAverage << 1) + newValue) / 3);
+        break;
+
+    case 3:
+        // New average = old average * 3/4 + new value * 1/4
+        mAverage = static_cast<uint16_t>(((oldAverage << 2) - oldAverage + newValue) >> 2);
+        break;
+
+    case 4:
+        // New average = old average * 4/5 + new value * 1/5
+        mAverage = static_cast<uint16_t>(((oldAverage << 2) + newValue) / 5);
+        break;
+
+    case 5:
+        // New average = old average * 5/6 + new value * 1/6
+        mAverage = static_cast<uint16_t>(((oldAverage << 2) + oldAverage + newValue) / 6);
+        break;
+
+    case 6:
```

Comment:
Thanks for updating this logic.  I understand that you are trying to optimize to use shifts and adds, rather than a single multiply.  Effectively trading code size for efficiency.  Is that the right tradeoff here?  `mCount < 8` only occurs when a link is first being established.  In steady state, we will be typically at `mCount >= 8`.